cabal-docspec/MANUAL.md

```diff
@@ -83,11 +83,32 @@ However, in this list we mostly only list and show the --option version of them.
 :   Message to return when the evaluation is timed out.
     Default is **\* Hangs forever \***.
 
+**\--ghci-rtsopts** *options*
+
+:   RTS options for GHCi process
+
+**\--skip-properties**
+
+:   Skip properties.
+
+**\--simple-properties**
+
+:   Evaluate **prop> expr x y** using **quickCheck (expr x y)**.
+    Requires *QuickCheck* package in the plan.
+
+**\--property-variables** *varlist*
+
+:   Variables to automatically quantify over in properties.
```

Comment:
Perhaps leave out the word "automatically". After all, the whole point of this flag is that you're _manually_ specifying the variable names, rather than having them be quantified over without input (i.e., automatically).

Also, what happens if a property variable happens to shadow the name of some other thing in scope? (Imagine `--property-variables=reverse`, for example.)

Reply:
I don't understand the question about shadowing.

Reply:
This works so when you have `--property-variables=xs` a property

```haskell
-- prop> sort xs == sort (sort xs)
```

is transformed into

```
quickCheck (\ xs -> sort xs == sort (sort xs))
```

expression, which is then run.

If we have `--property=variables="xs reverse"` and do

```haskell
-- prop> xs == reverse (reverse xs)
```

it will be transformed into

```haskell
quickCheck ( \ reverse xs -> xs == reverse (reverse xs))
```

which won't even run. (`No instance for (Show (() -> ()))`)

the idea is to look for symbols in the expression and quantify over ones which are in the `--property-variables` list. (A bit like implicit generalization with `variables` in Agda). The `exprVars` parser is simplistic, but seems toe enough. (Probably good idea to not do too clever shadowing in properties anyway).